File: 2021/fib/geeks-DPfib.c
#include <stdio.h>

long long fib(int n)
{
    long long f[n+2];
    int i;

    f[0] = 0;
    f[1] = 1;

    for (i = 2; i <= n; i++)
    {
        f[i] = f[i-1] + f[i-2];
    }

    return f[n];
}

int main()
{
    int n;
    scanf("%d", &n);
    printf("%lld\n", fib(n));
    return 0;
}

File: 2021/fib/geeks-DPfib.py
def fib(n):
    if n < 1: return -1
    f = []
    f.append(0)
    f.append(1)
    for i in range(2,n+1):
        f.append( f[i-1] + f[i-2] )
    return f[n-1]

print(fib(int(input())))

#for i in range(10):
#    print(fib(i))
